php/src/Google/Protobuf/Field_Cardinality.php

```diff
@@ -4,36 +4,7 @@
 
 namespace Google\Protobuf;
 
-/**
- * Whether a field is optional, required, or repeated.
- *
- * Protobuf enum <code>Google\Protobuf\Field\Cardinality</code>
- */
-class Field_Cardinality
-{
-    /**
-     * For fields with unknown cardinality.
-     *
-     * Generated from protobuf enum <code>CARDINALITY_UNKNOWN = 0;</code>
-     */
-    const CARDINALITY_UNKNOWN = 0;
-    /**
-     * For optional fields.
-     *
-     * Generated from protobuf enum <code>CARDINALITY_OPTIONAL = 1;</code>
-     */
-    const CARDINALITY_OPTIONAL = 1;
-    /**
-     * For required fields. Proto2 syntax only.
-     *
-     * Generated from protobuf enum <code>CARDINALITY_REQUIRED = 2;</code>
-     */
-    const CARDINALITY_REQUIRED = 2;
-    /**
-     * For repeated fields.
-     *
-     * Generated from protobuf enum <code>CARDINALITY_REPEATED = 3;</code>
-     */
-    const CARDINALITY_REPEATED = 3;
-}
+// This class has been renamed.
+// @see Google\Protobuf\Field\Cardinality
+class_exists(Field\Cardinality::class);
```

Comment:
Can we also add a deprecate warning here?

Reply:
Yeah, not a problem.

Reply:
One thing to consider about using `trigger_error` with the deprecated warning is that many systems wrap `E_USER_DEPRECATED` warnings in an exception. So this will break backwards compatibility for some users. 

Symfony gets around this by suppressing the warning, e.g. `@trigger_error`. This allows the deprecated warning to be detected but it will not break the application. Here are some examples:

 - [Guzzle](https://github.com/guzzle/guzzle/blob/ec70becf65a682371e9e10c3156eb01dc5fb4d26/src/Exception/BadResponseException.php#L20)
 - [Symfony](https://github.com/symfony/symfony/blob/61da48799c2ce020a72915311d053ec4557d6923/src/Symfony/Component/DependencyInjection/TypedReference.php#L32)
 - [Twig](https://github.com/twigphp/Twig/blob/ed1ce4d0d11abd68dd6e3f97d78ceb6e5ca238e5/lib/Twig/Function.php#L38)

I will trigger a silenced warning for now. LMK what you think.

Reply:
I was thinking just ```@deprecated```

Reply:
Okay, I will do that also. Let me know if you'd like me to remove the `@trigger_error`